### DISCUSSION TITLE
How to manage async data in next.config.js ?

### DISCUSSION THREAD

**Discussion by geosigno** [vercel/next.js#24457] [Help] [answered]
Here the thing, I want to use the next/image component for my images, but the images can come from different subdomains depending on which user is connecting.

So the idea is to dynamically fetch all the subdomains, and then populate the domain object in the next config.

Right now, I got this:

```
let imagePaths = []

let p = (async ()=>{
  const response = await axios({
    method: "post",
    url: 'API',
  });
  imagePaths = response.data.users.map(user => user.id+'.blob.core.windows.net')
})()
...
        images: {
          domains: imagePaths,
        },

```

But this is not working because the domains are fetched asynchronously. Ideally i would use "*.blob.core.windows.net" but this is not authorized.

Any idea?

**Comment by bobarros** [comment, marked as answer]
I liked the idea and tried to investigate it. This is what I did and worked for me. 

First, I build an external function to get async data. It runs before yarn dev, yarn build, etc. 
The result is a JSON file saved local that looks like this:

```json
{
  "domains": ["img.youtube.com","youtube.com"]
}
```

I also have a local function that loads and retrieves all domains:

```javascript
const imagesJSON = require("./imagesJSON.json")

const imageDomains = () => {
  return imagesJSON.domains;
}

module.exports = imageDomains;
```

Finally, my config file looks like this:

```javascript
const imageDomains = require("./functions/imageDomains");

// next.config.js
module.exports = {
  images: {
    domains: imageDomains(),
  }
};
```

I tried to use async functions all around and it didn't work. 
Maybe you can use this idea now and find a way to improve it.

I hope it works for you too.

**Reply by geosigno** [reply]
Thanks for that, it's work :)
